src/Confluent.Kafka/Consumer.cs

```diff
@@ -408,6 +408,11 @@ private void LogCallback(IntPtr rk, int level, string fac, string buf)
         /// </remarks>
         public Consumer(IEnumerable<KeyValuePair<string, object>> config)
         {
+            if (config.FirstOrDefault(prop => prop.Key == "group.id").Value == null)
```

Comment:
Thanks for the review Tomáš! I wasn't familiar with those recommendations. Yep, I agree it's better to follow convention here.